ip_list_test.go

```diff
@@ -0,0 +1,464 @@
+	}
+
+	actual, err := client.GetIPList(context.Background(), "2c0fc9fa937b11eaa1b71c4d701ab86e")
+	if assert.NoError(t, err) {
+		assert.Equal(t, want, actual)
+	}
+}
+
+func TestUpdateIPList(t *testing.T) {
+	setup(UsingAccount("foo"))
+	defer teardown()
+
+	handler := func(w http.ResponseWriter, r *http.Request) {
+		assert.Equal(t, r.Method, "PUT", "Expected method 'PUT', got %s", r.Method)
+		w.Header().Set("content-type", "application/json")
+		fmt.Fprint(w, `{
+			"result": {
+				"id": "2c0fc9fa937b11eaa1b71c4d701ab86e",
+				"name": "list1",
+				"description": "This note was updated.",
+				"kind": "ip",
+				"num_items": 10,
+				"num_referencing_filters": 2,
+				"created_on": "2020-01-01T08:00:00Z",
+				"modified_on": "2020-01-10T14:00:00Z"
+			},
+			"success": true,
+			"errors": [],
+			"messages": []
+		}`)
+	}
+
+	mux.HandleFunc("/accounts/foo/rules/lists/2c0fc9fa937b11eaa1b71c4d701ab86e", handler)
+
+	createdOn, _ := time.Parse(time.RFC3339, "2020-01-01T08:00:00Z")
+	modifiedOn, _ := time.Parse(time.RFC3339, "2020-01-10T14:00:00Z")
+
+	want := IPList{
+		ID:                    "2c0fc9fa937b11eaa1b71c4d701ab86e",
+		Name:                  "list1",
+		Description:           "This note was updated.",
+		Kind:                  "ip",
+		NumItems:              10,
+		NumReferencingFilters: 2,
+		CreatedOn:             &createdOn,
+		ModifiedOn:            &modifiedOn,
+	}
+
+	actual, err := client.UpdateIPList(context.Background(), "2c0fc9fa937b11eaa1b71c4d701ab86e",
+		"This note was updated.")
+	if assert.NoError(t, err) {
+		assert.Equal(t, want, actual)
+	}
+}
+
+func TestDeleteIPList(t *testing.T) {
+	setup(UsingAccount("foo"))
+	defer teardown()
+
+	handler := func(w http.ResponseWriter, r *http.Request) {
+		assert.Equal(t, r.Method, "DELETE", "Expected method 'DELETE', got %s", r.Method)
+		w.Header().Set("content-type", "application/json")
+		fmt.Fprint(w, `{
+			"result": {
+				"id": "34b12448945f11eaa1b71c4d701ab86e"
+			},
+			"success": true,
+			"errors": [],
+			"messages": []
+		}`)
+	}
+
+	mux.HandleFunc("/accounts/foo/rules/lists/2c0fc9fa937b11eaa1b71c4d701ab86e", handler)
+
+	want := IPListDeleteResponse{}
+	want.Success = true
+	want.Errors = []ResponseInfo{}
+	want.Messages = []ResponseInfo{}
+	want.Result.ID = "34b12448945f11eaa1b71c4d701ab86e"
+
+	actual, err := client.DeleteIPList(context.Background(), "2c0fc9fa937b11eaa1b71c4d701ab86e")
+	if assert.NoError(t, err) {
+		assert.Equal(t, want, actual)
+	}
+}
+
+func TestListIPListsItems(t *testing.T) {
+	setup(UsingAccount("foo"))
+	defer teardown()
+
+	handler := func(w http.ResponseWriter, r *http.Request) {
+		assert.Equal(t, r.Method, "GET", "Expected method 'GET', got %s", r.Method)
+		w.Header().Set("content-type", "application/json")
+
+		if len(r.URL.Query().Get("cursor")) > 0 && r.URL.Query().Get("cursor") == "yyy" {
+			fmt.Fprint(w, `{
+			"result": [
+    			{
+      				"id": "2c0fc9fa937b11eaa1b71c4d701ab86e",
+      				"ip": "10.0.0.2",
```

Comment:
Are we able to swap this IP out for [a reserved IP designated for documentation/examples](https://tools.ietf.org/html/rfc5737)?

```suggestion
      				"ip": "192.0.2.2",
```